scripts/cmake/z_vcpkg_fixup_rpath_macho.cmake

```
@@ -0,0 +1,154 @@
+    # remove trailing slash
+    string(REGEX REPLACE "/+$" "" relative_to_lib "${relative_to_lib}")
+
+    if(NOT relative_to_lib STREQUAL "")
+        set(new_rpath "@loader_path/${relative_to_lib}")
+    else()
+        set(new_rpath "@loader_path")
+    endif()
+
+    set("${arg_OUT_NEW_RPATH_VAR}" "${new_rpath}" PARENT_SCOPE)
+endfunction()
+
+function(z_vcpkg_fixup_macho_rpath_in_dir)
+    # We need to iterate through everything because we
+    # can't predict where a Mach-O file will be located
+    file(GLOB root_entries LIST_DIRECTORIES TRUE "${CURRENT_PACKAGES_DIR}/*")
+
+    # Skip some folders for better throughput
+    list(APPEND folders_to_skip "include")
+    list(JOIN folders_to_skip "|" folders_to_skip_regex)
+    set(folders_to_skip_regex "^(${folders_to_skip_regex})$")
+
+    find_program(
+        install_name_tool_cmd
+        NAMES install_name_tool
+        DOC "Absolute path of install_name_tool cmd"
+        REQUIRED
+    )
+
+    find_program(
+        otool_cmd
+        NAMES otool
+        DOC "Absolute path of otool cmd"
+        REQUIRED
+    )
+
+    find_program(
+        file_cmd
+        NAMES file
+        DOC "Absolute path of file cmd"
+        REQUIRED
+      )
+
+    foreach(folder IN LISTS root_entries)
+        if(NOT IS_DIRECTORY "${folder}")
+            continue()
+        endif()
+
+        get_filename_component(folder_name "${folder}" NAME)
+        if(folder_name MATCHES "${folders_to_skip_regex}")
+            continue()
+        endif()
+
+        file(GLOB_RECURSE macho_files LIST_DIRECTORIES FALSE "${folder}/*")
+        list(FILTER macho_files EXCLUDE REGEX "\\\.(cpp|cc|cxx|c|hpp|h|hh|hxx|inc|json|toml|yaml|man|m4|ac|am|in|log|txt|pyi?|pyc|pyx|pxd|pc|cmake|f77|f90|f03|fi|f|cu|mod|ini|whl|cat|csv|rst|md|npy|npz|template|build)$")
+        list(FILTER macho_files EXCLUDE REGEX "/(copyright|LICENSE|METADATA)$")
+
+        foreach(macho_file IN LISTS macho_files)
+            if(IS_SYMLINK "${macho_file}")
+                continue()
+            endif()
+
+            # Determine if the file is a Mach-O executable or shared library
+            execute_process(
+                COMMAND "${file_cmd}" -b "${macho_file}"
+                OUTPUT_VARIABLE file_output
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+            if(file_output MATCHES ".*Mach-O.*shared library.*")
+                set(file_type "shared")
+            elseif(file_output MATCHES ".*Mach-O.*executable.*")
+                set(file_type "executable")
+            else()
+                continue()
+            endif()
+
+            get_filename_component(macho_file_dir "${macho_file}" DIRECTORY)
+            get_filename_component(macho_file_name "${macho_file}" NAME)
+
+            z_vcpkg_calculate_corrected_macho_rpath(
+              MACHO_FILE_DIR "${macho_file_dir}"
+              OUT_NEW_RPATH_VAR new_rpath
+            )

```

```suggestion
            z_vcpkg_calculate_corrected_macho_rpath(
                MACHO_FILE_DIR "${macho_file_dir}"
                OUT_NEW_RPATH_VAR new_rpath
            )
```

It looks like this is a 4 spaces file? It is not worth making this change if no other edits to this file are needed.